src/AppBundle/Controller/HomepageController.php

```diff
@@ -11,42 +12,53 @@
  */
 class HomepageController extends Controller
 {
-
-	/**
-	 * @Route("/", name="homepage")
-	 * @param Request $request
-	 * @return \Symfony\Component\HttpFoundation\Response
-	 */
-	public function homepageAction(Request $request)
-	{
-		return $this->render("homepage/homepage.html.twig", [
-			"products" => $this->getDoctrine()->getRepository(Product::class)->findBy(
-				[],
-				[
-					"rank" => "desc"
-				],
-				20
-			),
-			"categories" => $this->getDoctrine()->getRepository(Category::class)->findBy(
-				[],
-				[
-					"rank" => "desc",
-				]
-			),
-		]);
-	}
-
-
-	/**
-	 * @Route("/hello", name="hello")
-	 * @param Request $request
-	 * @return \Symfony\Component\HttpFoundation\Response
-	 */
-	public function helloAction(Request $request)
-	{
-		return $this->render("homepage/hello.html.twig", [
-			"who" => "world",
-		]);
-	}
-
+    
+    /**
+     * @Route("/", name="homepage")
+     * @param Request $request
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function homepageAction(Request $request)
+    {
+    
+        $paginator  = $this->get('knp_paginator');
+        $em = $this->getDoctrine()->getManager();
+        $all = $em->getRepository(Product::class)->findAll();
```

Comment:
Tady pozor tohle ti vytáhne celou tabulku do paměti. To není zrovna efektivní. Při pár tisících položek a pár requestech ti umře jakýkoliv server. :)